src/main/java/org/PackageDel/Delivery.java

```diff
@@ -0,0 +1,67 @@
+package org.PackageDel;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Delivery {
+
+    private static final List<Package> packages = new ArrayList<>();
+
+    public static void main(String[] args) throws Exception {
+        loadPackagesFromFile();
+
+        Map<String, Map<LocalDate, List<Package>>> grouped = packages.stream()
+                .collect(Collectors.groupingBy(Package::getLocation,
+                        Collectors.groupingBy(Package::getDeliveryDate)));
+
+        int totalValue = 0;
+        int totalRevenue = 0;
+
+        List<Thread> threads = new ArrayList<>();
+
+        for (var entry : grouped.entrySet()) {
+            String location = entry.getKey();
+            for (var dateEntry : entry.getValue().entrySet()) {
+                LocalDate date = dateEntry.getKey();
+                List<Package> groupPackages = dateEntry.getValue();
+                int distance = groupPackages.get(0).getDistanceKm();
+                PackageGroup group = new PackageGroup(location, date, distance, groupPackages);
+                totalValue += group.getGroupValue();
+                totalRevenue += group.getGroupRevenue();
+
+                DeliveryTask task = new DeliveryTask(group);
+                threads.add(task);
+                task.start();
+            }
+        }
+
+        for (Thread t : threads) {
+            t.join();
+        }
+
+        System.out.println("\nRESULT:");
+        System.out.println("Total value of all delivered packages: " + totalValue + " LEI");
+        System.out.println("Total delivery revenue: " + totalRevenue + " LEI");
+    }
+
+    private static void loadPackagesFromFile() throws Exception {
```

Comment:
better to delegate this to a specialised PacketFileLoader class. Additionally, better to have it as 
```suggestion
    private List<Package> loadPackagesFromFile() throws Exception {
```
for better reusability and keepint this method similar to a pure function (returning a list instead of mutating the state of the main class)